internal/graphicsdriver/metal/mtl/mtl_darwin.go

```diff
@@ -849,103 +1024,130 @@ func (l Library) MakeFunction(name string) (Function, error) {
 //
 // Reference: https://developer.apple.com/documentation/metal/mtltexture.
 type Texture struct {
-	texture unsafe.Pointer
+	texture objc.ID
 }
 
 // NewTexture returns a Texture that wraps an existing id<MTLTexture> pointer.
-func NewTexture(texture unsafe.Pointer) Texture {
+func NewTexture(texture objc.ID) Texture {
 	return Texture{texture: texture}
 }
 
 // resource implements the Resource interface.
-func (t Texture) resource() unsafe.Pointer { return t.texture }
+func (t Texture) resource() unsafe.Pointer { return *(*unsafe.Pointer)(unsafe.Pointer(&t.texture)) }
 
 func (t Texture) Release() {
-	C.Texture_Release(t.texture)
+	objc.ID(t.texture).Send(sel_release)
 }
 
 // GetBytes copies a block of pixels from the storage allocation of texture
 // slice zero into system memory at a specified address.
 //
 // Reference: https://developer.apple.com/documentation/metal/mtltexture/1515751-getbytes.
 func (t Texture) GetBytes(pixelBytes *byte, bytesPerRow uintptr, region Region, level int) {
-	r := region.c()
-	C.Texture_GetBytes(t.texture, unsafe.Pointer(pixelBytes), C.size_t(bytesPerRow), r, C.uint_t(level))
+	inv := cocoa.NSInvocation_invocationWithMethodSignature(cocoa.NSMethodSignature_SignatureWithObjCTypes("v@:^vQ{MTLRegion={MTLOrigin=qqq}{MTLSize=qqq}}Q"))
```

Comment:
Just out of curiosity, is there a document about this signature string rule?

Reply:
Yeah it’s what’s returned from [@encode](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/ObjCRuntimeGuide/Articles/ocrtTypeEncodings.html#//apple_ref/doc/uid/TP40008048-CH100). Obviously the best thing is to get the signature from the method itself but that doesn’t work with protocols bc the only way is with the protocol_getDescription function which returns a struct. We don’t have struct return support yet which is the reason I’m using NSInvocation.

Reply:
Thanks!

Reply:
> We don’t have struct return support yet which is the reason I’m using NSInvocation.

I think this is a good point to leave a comment.

Reply:
Sure I leave it by use of nsinvocation or just at its definition?

Reply:
just at its definition seems enough, thanks!